### PR TITLE
Singleton Interface

### PR DESCRIPTION
**Pull request by volodymyr-mykhailyk**
Change interface of generator to singelton

### CODE REVIEW

File: lib/id_generator.rb [hunk 1]
```diff
@@ -1,46 +1,22 @@
...
-
-    true
+require 'id_generator/version'
+require 'id_generator/configuration'
+require 'id_generator/errors'
+require 'id_generator/generators/timestamped_random'
+
+module IdGenerator
+  def self.configure
+    yield(configuration)
```
Comment (Fattaf) [inline lib/id_generator.rb:8]: You can use here `Singleton` core ruby module, as a good alternative.

File: lib/id_generator/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module IdGenerator
+  class Configuration
+    attr_reader :context_id
+
+    def initialize(context_id: 0)
```
Comment (Fattaf) [inline lib/id_generator/configuration.rb:5]: Are you sure you want to default it here?

**Review by Fattaf — APPROVED** [review]

File: lib/id_generator/configuration.rb (hunk 2 above)
Comment (volodymyr-mykhailyk) [reply lib/id_generator/configuration.rb:5]: @Fattaf Let's leave it for now. My thought process was that configuration and defaults leave in one place. I agree that if we have more - we can move it somewhere

File: lib/id_generator.rb (hunk 1 above)
Comment (volodymyr-mykhailyk) [reply lib/id_generator.rb:8]: Can you use Singleton for modules? In this case, gems are usually designed in a way that you have a top module with configuration that using instances defined inside of a gem

**Review by vnovitskyi — APPROVED** [review]
Nice work

**Pull request merged**